### PR TITLE
Datacollector env variables should be given more priority.

### PR DESCRIPTION
**Pull request by smadala**
- When Datacollector want to set env variables and if same env variable already set in system level. Datacollector value should be given more priority.

### CODE REVIEW

**Review by mayankbansal018 — COMMENTED** [review]
Changes look good, but any particular reason why we are not adding test case now?

File: src/Microsoft.TestPlatform.PlatformAbstractions/net451/System/ProcessStartInfoExtensions.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ public static class ProcessStartInfoExtensions
         /// <param name="value">Environment Variable value.</param>
         public static void AddEnvironmentVariable(this ProcessStartInfo startInfo, string name, string value)
         {
-            startInfo.EnvironmentVariables.Add(name, value);
+            startInfo.EnvironmentVariables[name] = value;
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.PlatformAbstractions/net451/System/ProcessStartInfoExtensions.cs]: Have we tested that this doesn't throw exception on overriding?
Earlier even add was not throwing exception.

**Review by mayankbansal018 — APPROVED** [review]

**Review by codito — CHANGES_REQUESTED** [review]
Require tests

File: src/Microsoft.TestPlatform.PlatformAbstractions/net451/System/ProcessStartInfoExtensions.cs (hunk 1 above)
Comment (smadala) [inline src/Microsoft.TestPlatform.PlatformAbstractions/net451/System/ProcessStartInfoExtensions.cs]: Tested it. Worked as expected.

**Review by codito — APPROVED** [review]

**Pull request merged**